eventstore/views.py

```diff
@@ -20,6 +27,89 @@
     PrebirthRegistrationSerializer,
     PublicRegistrationSerializer,
 )
+from ndoh_hub.utils import validate_signature
+
+
+class MessagesViewSet(GenericViewSet):
+    def create(self, request):
+        validate_signature(request)
+        try:
+            webhook_type = request.META.get("HTTP_X_TURN_HOOK_SUBSCRIPTION", "whatsapp")
+            if webhook_type == "whatsapp":
+                for inbound in request.data.get("messages", []):
+                    data = {
+                        "context": inbound.get("context"),
+                        "errors": inbound.get("errors"),
+                        "audio": inbound.get("audio"),
+                        "document": inbound.get("document"),
+                        "image": inbound.get("image"),
+                        "system": inbound.get("system"),
+                        "location": inbound.get("location"),
+                        "text": inbound.get("text"),
+                        "video": inbound.get("video"),
+                        "voice": inbound.get("voice"),
+                    }
```

Comment:
Doing it this way won't allow us to easily adapt to new fields being added to the message body, we would have to change this block of code to be able to adapt to changes in the webhook.

Perhaps a better way of doing this is to use Python's `pop` method on the dictionary, ie. For each of the required fields, we do a `pop`
```
id = inbound.pop("id")
contact_id = inbound.pop("from")
...
```
Then, anything remaining in `inbound`, we can put into `data` on the `Messages`, ie.
```
Messages.objects.create(
  id=id,
  contact_id=contact_id,
  ...,
  data=inbound
)
```

Reply:
will try this out